yt/utilities/io_handler.py

```diff
@@ -12,6 +13,26 @@
 
 use_caching = 0
 
+FluidFieldType = str
+FluidFieldName = str
+FluidFieldTuple = typing.Tuple[FluidFieldType, FluidFieldName]
+
+ParticleFieldType = str
+ParticleFieldName = str
+ParticleFieldTuple = typing.Tuple[ParticleFieldType, ParticleFieldName]
+
+ParticleTypeFields = typing.Dict[ParticleFieldType, typing.List[ParticleFieldName]]
+ParticleTypeSizes = typing.Dict[ParticleFieldType, int]
+ParticleFieldSize = typing.Dict[ParticleFieldTuple, int]
+ParticleCoordinateSet = typing.Tuple[
+    str,
+    typing.Tuple[np.ndarray, np.ndarray, np.ndarray],
+    typing.Union[float, np.ndarray],
```

Comment:
this line is probably not as general as necessary. I bet any real number (`from numbers import Real`) is valid, which includes numpy floating and integer scalar types. A List should probably be valid too where a np.ndarray works, so this single line should probably have a type of its own